### PR TITLE
chore(core): add general builder and cli-fixture tests

### PR DESCRIPTION
**Pull request by JoviDeCroock**
Resolves https://github.com/StellateHQ/fuse.js/issues/95

This adds a set of builder tests of the parts that we made custom and perform more logic than the battle-tested [pothos](https://github.com/hayes/pothos/tree/main/packages/core/tests). We also add general tests for our `cli` in terms of dev expectations

### CODE REVIEW

**Comment by mxstbr** [comment]
Love it! 🙌

File: packages/core/test/list.test.ts [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+  expect(printSchema(schema)).toMatchInlineSnapshot(`
+    """"
+    A date string, such as 2007-12-03, compliant with the \`full-date\` format outlined in section 5.6 of the RFC 3339 profile of the ISO 8601 standard for representation of dates and times using the Gregorian calendar.
+    """
+    scalar Date
+
+    """
+    The \`JSON\` scalar type represents JSON values as specified by [ECMA-404](http://www.ecma-international.org/publications/files/ECMA-ST/ECMA-404.pdf).
+    """
+    scalar JSON
```
Comment (danstarns) [inline packages/core/test/list.test.ts:148]: How come Date and JSON are always included in the generated schema even if they aren't used?

**Review by danstarns — COMMENTED** [review]
👀

File: packages/core/test/list.test.ts (hunk 1 above)
Comment (JoviDeCroock) [reply packages/core/test/list.test.ts:148]: Because they are added as a scalar, by default gql does no pruning of unused types

Comment (danstarns) [reply packages/core/test/list.test.ts:148]: Do think it is worth pruning them if they are not used? I can help with this if so.

**Pull request merged**